aten/src/ATen/native/TensorShape.cpp

```diff
@@ -425,8 +426,9 @@ static inline Tensor & sparse_transpose_(Tensor & self, int64_t dim0, int64_t di
     auto sizes = self.sizes().vec();
     std::swap(sizes[dim0], sizes[dim1]);
 
-    return self.sparse_resize_(sizes, self._indices().size(0), self._values().dim() - 1);
+    _get_sparse_impl(self)->raw_resize_(self._indices().size(0), self._values().dim() - 1, sizes);
```

Comment:
Yes it's because a lot of times when doing transpose on sparse dimensions we are effectively shrinking one of them (and expanding another), and it's inherently an unsafe operation.